### PR TITLE
25150 add new srcset usage example

### LINKED ISSUE #25150: A srcset example contradicts instructions
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture

### What specific section or headline is this issue about?

The srcset attribute

### What information was incorrect, unhelpful, or incomplete?

Earlier on the page, the reader is told:

> If providing higher-density versions of an image for high-DPI (Retina) display, use [srcset](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-srcset) on the `` element instead. This lets browsers opt for lower-density versions in data-saving modes, and you don't have to write explicit media conditions.

But in "The srcset attribute" section, the example used to demonstrate how to include alternate images based on pixel density only uses `srcset` in the `<source>` elements and `src` in the `` element.

### What did you expect to see?

I expected the example to use `srcset` in the same way the page instructed earlier, in the `` element.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/element/picture`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/picture/index.md
* Last commit: https://github.com/mdn/content/commit/b24feefa4d8e807f77abe935d33f49fc92f84921
* Document last modified: 2023-03-03T04:26:54.000Z

</details>

### PR DESCRIPTION
**Pull request by adarshajit**
### Description

This PR adds an new example of using `srcset` attribute on the `` tag. 

### Motivation

Earlier on the page, the reader is told:

> If providing higher-density versions of an image for high-DPI (Retina) display, use [srcset](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-srcset) on the  element instead. This lets browsers opt for lower-density versions in data-saving modes, and you don't have to write explicit media conditions.

But in "The srcset attribute" section, the example used to demonstrate how to include alternate images based on pixel density only uses srcset in the <source> elements and src in the  element.

### Additional details

NIL

### Related issues and pull requests

Fixes #25150

### CODE REVIEW

**Comment by schalkneethling** [comment]
Hey @adarshajit 👋 Thank you for your pull request. Before I start a review, please refer to at the Markdown lint error that the CI is raising. Thanks.

```
Finding: files/en-us/web/html/element/picture/index.md !node_modules !.github !tests
Linting: 1 file(s)
Warning: files/en-us/web/html/element/picture/index.md:[14](https://github.com/mdn/content/actions/runs/4902351202/jobs/8756077877?pr=26618#step:6:15)3 MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
Summary: 1 error(s)
```

File: files/en-us/web/html/element/picture/index.md [hunk 1]
```diff
@@ -117,13 +117,37 @@ It is composed of a comma-separated list of image descriptors. Each image descri
   _OR_
 - a _pixel density descriptor_, followed by an `x` (such as `2x`) to serve a high-res image for high-DPI screens.
 
+This is an example on how to use `srcset` with a `<picture>` tag:
```
Comment (schalkneethling) [inline files/en-us/web/html/element/picture/index.md:120]: How about we reword it to the following?

```suggestion
The following example demonstrates how to use the `srcset` attribute with the `<source>` element to specify a high-density and standard-resolution image:
```

On a side note, on line 112 there is a small typo that would be great to fix in this pull request.

Currently:
```markdown
The [srcset](/en-US/docs/Web/HTML/Element/source#srcset) attribute is used to offer list of possible images _based on size_.
```

Correct:
```markdown
The [srcset](/en-US/docs/Web/HTML/Element/source#srcset) attribute is used to offer a list of possible images _based on size_.
```

**Review by schalkneethling — CHANGES_REQUESTED** [review]
Hey, @adarshajit. I have added a few comments and suggestions. Let me know what you think. Thank you for your contribution and patience.

**Comment by schalkneethling** [comment]
Hey, @adarshajit, no rush, but if any of my suggestions are unclear, please let me know. Thanks!

File: files/en-us/web/html/element/picture/index.md [hunk 2]
```diff
@@ -109,21 +109,42 @@ If the {{HTMLElement("source")}}'s media condition evaluates to `false`, the bro
...
+  width="320"
+  alt="MDN Web Docs logo"
+/>
+```
+
+The `sizes` attribute is not mandatory when using srcset, but it is recommended to use it in order to provide additional information to the browser to help it select the best image source.
+
+Without sizes, the browser will use the default size of the image as specified by its dimensions in pixels. This may not be the best fit for all devices, especially if the image is displayed on different screen sizes or in different contexts.
+
+Please note that sizes will have its effect only if width dimension descriptors are provided with srcset instead of pixel ratio values (200w instead of 2x for example).
```
Comment (schalkneethling) [inline files/en-us/web/html/element/picture/index.md:146]: We can remove all of this as there is no mention of the `sizes` attribute on this page.

File: files/en-us/web/html/element/picture/index.md [hunk 3]
```diff
@@ -109,21 +109,42 @@ If the {{HTMLElement("source")}}'s media condition evaluates to `false`, the bro
 
 ### The srcset attribute
 
-The [srcset](/en-US/docs/Web/HTML/Element/source#srcset) attribute is used to offer list of possible images _based on size_.
+The [srcset](/en-US/docs/Web/HTML/Element/source#srcset) attribute is used to offer a list of possible images _based on size_.
```
Comment (schalkneethling) [inline files/en-us/web/html/element/picture/index.md:112]: This is not something you added or changed, but I am not sure "based on size" is entirely accurate. You can specify different images using a width descriptor but also a pixel density descriptor. Pixel density is not really related to size. You can have a mobile phone with a high-density display just as you can have a laptop with a high-density display. @bsmth, what do you think?

File: files/en-us/web/html/element/picture/index.md [hunk 4]
```diff
@@ -109,21 +109,42 @@ If the {{HTMLElement("source")}}'s media condition evaluates to `false`, the bro
...
+The following example illustrates the usage of `srcset` attribute with the `<source>` element to specify a high-density and standard-resolution image:
+
 ```html
 <picture>
-  <source srcset="logo-768.png, logo-768-1.5x.png 1.5x" />
-  <img src="logo-320.png" alt="logo" />
+  <source srcset="logo.png, 
+                  logo-1.5x.png 1.5x" />
+  <img src="logo.png" alt="MDN Web Docs logo" height="320" width="320" />
 </picture>
```
Comment (schalkneethling) [inline files/en-us/web/html/element/picture/index.md:127]: Looking at some of the other content on this page, I am rethinking the code examples. As the main use case for the `picture` element is art direction and offering different image formats, it might be a good idea to either use the code from the interactive example at the top of the page (this one demonstrates art direction):

```html
<picture>
    <source srcset="/media/cc0-images/surfer-240-200.jpg"
            media="(orientation: portrait)">
    <img src="/media/cc0-images/painted-hand-298-332.jpg" alt="">
</picture>
```

Or the following, which demonstrates how you can offer different image formats and fallback to PNG:

```html
<picture>
  <source srcset="trevor-gerzen-r_XS7sml57o-unsplash.avif" type="image/avif" />
  <source srcset="trevor-gerzen-r_XS7sml57o-unsplash.webp" type="image/webp" />
  <img
    src="trevor-gerzen-r_XS7sml57o-unsplash.png"
    width="1223"
    height="815"
    alt="A beautiful sunset in the background with two surfers paddling into the surf in the foreground"
  />
</picture>
```

The nice thing about the second example is that when combined with the interactive example, the page demonstrated the two attributes unique to the `source` element, namely `media` and `type`. The rest of the attributes (`srcset`, `sizes`, `width`, and `height`) can also be used on the `img` element, as shown in the example you added further down. What do you think?

**Review by schalkneethling — CHANGES_REQUESTED** [review]
Hey @adarshajit, thank you for your patience and for sticking with this one as we figure out the best way to document this. I have made some suggestions for you to consider and also asked Brian to take a look.

File: files/en-us/web/html/element/picture/index.md (hunk 3 above)
Comment (jfrech) [reply files/en-us/web/html/element/picture/index.md:112]: I would have read it to mean "based on the size in true pixels which is the best fit". So changing "based on size" to e.&hairsp;g. "based on required resolution" would have my blessing.

**Comment by bsmth** [comment]
hi @adarshajit - thanks for the submission. I'll take a look at your changes shortly 👀

File: files/en-us/web/html/element/picture/index.md [hunk 5]
```diff
@@ -109,21 +109,42 @@ If the {{HTMLElement("source")}}'s media condition evaluates to `false`, the bro
...
 
+The following example illustrates the usage of `srcset` attribute with the `<source>` element to specify a high-density and standard-resolution image:
+
 ```html
 <picture>
-  <source srcset="logo-768.png, logo-768-1.5x.png 1.5x" />
-  <img src="logo-320.png" alt="logo" />
+  <source srcset="logo.png, 
+                  logo-1.5x.png 1.5x" />
+  <img src="logo.png" alt="MDN Web Docs logo" height="320" width="320" />
```
Comment (bsmth) [inline files/en-us/web/html/element/picture/index.md:126]: Re-adding Schalk's suggestion here:
```suggestion
  <source srcset="logo.png, 
                  logo@1.5x.png 1.5x" />
  <img src="logo.png" height="320" width="320" alt="logo" />
```

with this rationale:

>The reason for the above suggestion is to show that the  element acts as a fallback for browsers that do not support the <source> element. In this example, the <source> element specifies two images. One that will be used for high-definition displays and another for all others. Seeing that both the source of the  element and the first entry in the srcset attribute use the same file format, the addition of the size (-320 and -768) is not needed and might cause confusion for some readers.

I also added the height and width attributes so that we keep with best practices. Let me know what you think.

File: files/en-us/web/html/element/picture/index.md (hunk 3 above)
Comment (bsmth) [reply files/en-us/web/html/element/picture/index.md:112]: Schalk is right here, it's not just image size, it's also pixel density.

```suggestion
The [srcset](/en-US/docs/Web/HTML/Element/source#srcset) attribute is used to offer a list of possible images based on size or the display's pixel density.
```

See:

* https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/srcset
* https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images#resolution_switching_same_size_different_resolutions
* https://html.spec.whatwg.org/multipage/embedded-content.html#the-source-element

File: files/en-us/web/html/element/picture/index.md (hunk 2 above)
Comment (bsmth) [reply files/en-us/web/html/element/picture/index.md:144]: I think it's okay to mention sizes here if we're mentioning `srcset`. I'm going to mark this as resolved for now

File: files/en-us/web/html/element/picture/index.md [hunk 6]
```diff
@@ -109,21 +109,40 @@ If the {{HTMLElement("source")}}'s media condition evaluates to `false`, the bro
...
 ### The srcset attribute
 
-The [srcset](/en-US/docs/Web/HTML/Element/source#srcset) attribute is used to offer list of possible images _based on size_.
+The [srcset](/en-US/docs/Web/HTML/Element/source#srcset) attribute is used to offer a list of possible images _based on size_.
 
 It is composed of a comma-separated list of image descriptors. Each image descriptor is composed of a URL of the image, and _either_:
 
 - a _width descriptor_, followed by a `w` (such as `300w`);
   _OR_
 - a _pixel density descriptor_, followed by an `x` (such as `2x`) to serve a high-res image for high-DPI screens.
```
Comment (bsmth) [inline files/en-us/web/html/element/picture/index.md:118]: I think as a follow-up to this PR, there's a couple of things to note:

* You can't mix with and pixel density descriptors
* missing pixel density descriptor implies `1x`.
* duplicate descriptor values are not allowed (`2x` & `2x`, `100w` & `100w`)

But we also have a doc about this here:

https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images#resolution_switching_same_size_different_resolutions

**Comment by bsmth** [comment]
> Please take a look when you have some time.

Thanks a lot, I'll take another look shortly 👀

**Review by bsmth — APPROVED** [review]
Thanks a lot for the submission @adarshajit - I think we can merge this one 👍🏻
